Alliance.Server/Patch/Behaviors/AllianceLobbyComponent.cs

```diff
@@ -3,57 +3,66 @@
+			MissionScoreboardComponent _missionScoreboardComponent = Mission.GetMissionBehavior<MissionScoreboardComponent>();
+			MissionScoreboardComponent.MissionScoreboardSide sideSafe = _missionScoreboardComponent.GetSideSafe(botAgent.Team.Side);
 
-            if (botAgent.Formation?.PlayerOwner != null)
-            {
-                NetworkCommunicator networkCommunicator = GameNetwork.NetworkPeers.SingleOrDefault((x) => x.GetComponent<MissionPeer>() != null && x.GetComponent<MissionPeer>().ControlledFormation == botAgent.Formation);
-                if (networkCommunicator != null)
-                {
-                    MissionPeer component = networkCommunicator.GetComponent<MissionPeer>();
-                    typeof(MissionPeer).GetProperty(nameof(MissionPeer.DeathCount)).SetValue(component, component.DeathCount + 1);
-                    // Check that BotsUnderControlAlive doesn't go under 0 to prevent client crash
-                    component.BotsUnderControlAlive = Math.Max(component.BotsUnderControlAlive - 1, 0);
-                    _missionScoreboardComponent.PlayerPropertiesChanged(networkCommunicator);
-                    GameNetwork.BeginBroadcastModuleEvent();
-                    GameNetwork.WriteMessage(new KillDeathCountChange(networkCommunicator, affectorPeer?.GetNetworkPeer(), component.KillCount, component.AssistCount, component.DeathCount, component.Score));
-                    GameNetwork.EndBroadcastModuleEvent(GameNetwork.EventBroadcastFlags.None);
-                    GameNetwork.BeginBroadcastModuleEvent();
-                    GameNetwork.WriteMessage(new BotsControlledChange(networkCommunicator, component.BotsUnderControlAlive, component.BotsUnderControlTotal));
-                    GameNetwork.EndBroadcastModuleEvent(GameNetwork.EventBroadcastFlags.None);
-                }
-            }
-            else
-            {
-                MissionScoreboardComponent.MissionScoreboardSide sideSafe = _missionScoreboardComponent.GetSideSafe(botAgent.Team.Side);
-                TaleWorlds.MountAndBlade.BotData botScores = sideSafe.BotScores;
-                botScores.DeathCount++;
-                botScores.AliveCount--;
-                GameNetwork.BeginBroadcastModuleEvent();
-                GameNetwork.WriteMessage(new NetworkMessages.FromServer.BotData(sideSafe.Side, botScores.KillCount, botScores.AssistCount, botScores.DeathCount, botScores.AliveCount));
-                GameNetwork.EndBroadcastModuleEvent(GameNetwork.EventBroadcastFlags.None);
-            }
+			TaleWorlds.MountAndBlade.BotData botScores = sideSafe.BotScores;
 
-            _missionScoreboardComponent.BotPropertiesChanged(botAgent.Team.Side);
-        }
-    }
+			// Remove bot from alive count in every case
+			botScores.AliveCount--;
+			if (botScores.AliveCount >= 2048 || botScores.AliveCount < 0)
+			{
+				Log($"ERROR in AllianceLobbyComponent : AliveCount out of bounds({botScores.AliveCount})", LogLevel.Error);
```

Comment:
Isn't the out of bounds can cause crashes ?
Can't we cap the values to theses boundaries if it exceed ?

Reply:
It isn't throwing a real out of bounds exception, it's just a compression info error when trying to send the GameNetwork message. The issue came from the AliveCount not being decremented in some cases, it should now be fine. I left the error log in case.